components/merkledb/src/proof_list_index/mod.rs

```diff
@@ -431,11 +433,13 @@ where
             )
         }
 
-        if to > self.len() {
-            ListProof::Absent(ProofOfAbsence::new(self.len(), self.merkle_root()))
+        let proof = if to > self.len() {
```

Comment:
Semi-offtopic: I find it slightly weird / unexpected that in case the range proof is *partially* valid (e.g., we request elements `5..10` in the three with 8 elements), we get no elements in return.

Reply:
If I understand correctly, it won't require anything atop of a proof for the present elements (as it already includes the length, which affects the list state hash, and allows to infer that the other elements are absent).

Reply:
I thought about creating a separate task to modify range proofs of absence to accommodate existing elements. But is it relevant in the case of new flat proofs proposal?

Reply:
IMO, it's an orthogonal to the internal proof organization, so you can still create the corresponding task 😉